### PR TITLE
DOC: spelling and format fixes

### PR DESCRIPTION
**Pull request by dbkinder**
Fixed some spelling errors and some edits for clarity and formatting

[skip ci]

Signed-off-by: David Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: doc/architecture/system-and-security-architecture.rst [hunk 1]
```diff
@@ -417,20 +417,20 @@ For a detailed discussion of firewall configuration, see
 Production and Development Images
 =================================
 
-By default, building an image results in something that is locked-down
-and secure. This is how real products should be built. Unless some
-kind of application gets installed during image creation, one cannot
-do much with the running image (no user interface, no way to log into
-the system).
+By default, real product images should be built locked-down
```
Comment (pohly) [inline doc/architecture/system-and-security-architecture.rst]: I think this change looses one important part of the message, which is that compiling Ostro OS results in secure images by default. I.e. don't change the configuration and you are safe.

The proposed wording ("product images should be built locked-down") makes it sound either like we are not sure whether they really are locked-down or that this merely a recommendation.

**Comment by pohly** [comment]
Looks good except for the part about "should be built locked-down".

File: doc/architecture/system-and-security-architecture.rst (hunk 1 above)
Comment (dbkinder) [inline doc/architecture/system-and-security-architecture.rst]: OK.  Thanks for the comment.  Sometimes editing to make something read better can, like this, subtly alter the meaning.  I'll try again...

**Comment by dbkinder** [comment]
I fixed the "should be built locked-down" ...

**Pull request merged**